rclcpp_lifecycle/include/rclcpp_lifecycle/lifecycle_node.hpp

```diff
@@ -280,6 +294,19 @@ class LifecycleNode : public node_interfaces::LifecycleNodeInterface,
     CallbackT callback,
     rclcpp::CallbackGroup::SharedPtr group = nullptr);
 
+  /// Create a lifecycle timer that uses the node clock to drive the callback.
```

Comment:
we should also have an API to create a timer with a custom clock

Reply:
I'm not sure whether that makes sense.  Sure, the underlying https://github.com/ros2/rclcpp/blob/rolling/rclcpp/include/rclcpp/create_timer.hpp has support for that, but the `rclcpp::Node` already has a `Clock` that is accessed via `get_clock`.  Giving the ability to create a timer based off another clock means that this particular timer will (possibly) run at a different rate from the rest of the `Node`.  While I can imagine some uses for that, it is kind of strange.

Further, it doesn't look like we expose that kind of functionality from https://github.com/ros2/rclcpp/blob/rolling/rclcpp/include/rclcpp/node.hpp .  And I don't really want `LifecycleNode` to "get ahead" of `Node` (we already have the opposite problem).

To me, adding that in is an orthogonal discussion, and if we are going to do it we should do it for both `Node` and `LifecycleNode`.